### PR TITLE
Fix issue of opening too many files in BlockBasedTableReaderCapMemoryTest.CapMemoryUsageUnderCacheCapacity

### PR DESCRIPTION
**Pull request by hx235**
**Context:** 
Unit test for https://github.com/facebook/rocksdb/pull/9748 keeps opening new files to see whether the new feature is able to correctly constrain the opening based on block cache capacity.

However, the unit test has two places written inefficiently that can lead to opening too many new files relative to underlying operating system/file system constraint, even before hitting the block cache capacity:
(1) [opened_table_reader_num < 2 * max_table_reader_num](https://github.com/facebook/rocksdb/pull/9748/files?show-viewed-files=true&file-filters%5B%5D=#diff-ec9f5353e317df71093094734ba29193b94a998f0f9c9af924e4c99692195eeaR438), which can leads to 1200 + open files because of (2) below
(2) NewLRUCache(6 * CacheReservationManagerImpl<CacheEntryRole::kBlockBasedTableReader>::GetDummyEntrySize()) in [here](https://github.com/facebook/rocksdb/pull/9748/files?show-viewed-files=true&file-filters%5B%5D=#diff-ec9f5353e317df71093094734ba29193b94a998f0f9c9af924e4c99692195eeaR364) 

Therefore we see CI failures like this on machine with a strict open file limit ~1000 (see the "table_1021" naming in following error msg)
https://app.circleci.com/pipelines/github/facebook/rocksdb/12886/workflows/75524682-3fa4-41ee-9a61-81827b51d99b/jobs/345270
```
fs_->NewWritableFile(path, foptions, &file, nullptr)
IO error: While open a file for appending: /dev/shm/rocksdb.Jedwt/run-block_based_table_reader_test-CapMemoryUsageUnderCacheCapacity-BlockBasedTableReaderCapMemoryTest.CapMemoryUsageUnderCacheCapacity-0/block_based_table_reader_test_1668910_829492452552920927/**table_1021**: Too many open files
```

**Summary:**
- Revised the test more efficiently on the above 2 places,  including using 1.1 instead 2 in the threshold and lowering down the block cache capacity a bit
- Renamed some variables for clarity

**Test plan:**
- Manual inspection of max opened table reader in all test case, which is around ~389
- Circle CI to see if error is gone

### CODE REVIEW

File: table/block_based/block_based_table_reader_test.cc [hunk 1]
```diff
@@ -449,31 +459,28 @@ TEST_P(BlockBasedTableReaderCapMemoryTest, CapMemoryUsageUnderCacheCapacity) {
   }
 
   if (reserve_table_reader_memory_) {
-    EXPECT_TRUE(s.IsMemoryLimit() &&
-                opened_table_reader_num < 2 * max_table_reader_num)
-        << "s: " << s.ToString() << " opened_table_reader_num: "
```
Comment (hx235) [inline table/block_based/block_based_table_reader_test.cc:454]: This is verify below in line 465 - 466 already therefore being removed here

**Review by ajkr — APPROVED** [review]

File: table/block_based/block_based_table_reader_test.cc [hunk 2]
```diff
@@ -420,22 +420,32 @@ class BlockBasedTableReaderCapMemoryTest
...
 
   // Keep creating BlockBasedTableReader till hiting the memory limit based on
-  // cache capacity and creation fails or reaching a big number of table readers
-  while (s.ok() && opened_table_reader_num < 2 * max_table_reader_num) {
+  // cache capacity and creation fails (when reserve_table_reader_memory_ ==
+  // true) or reaching a specfied big number of table readers (when
+  // reserve_table_reader_memory_ == false)
+  std::size_t max_table_reader_num_uncapped =
+      max_table_reader_num_capped + max_table_reader_num_capped / 10;
+  ASSERT_GT(max_table_reader_num_uncapped, max_table_reader_num_capped)
```
Comment (ajkr) [inline table/block_based/block_based_table_reader_test.cc:444]: Should it be `ASSERT_GT(max_table_reader_num_uncapped, max_table_reader_num_capped * 1.01)` to ensure we definitely opened more files in uncapped mode?

**Review by ajkr — COMMENTED** [review]
LGTM!

File: table/block_based/block_based_table_reader_test.cc (hunk 2 above)
Comment (hx235) [reply table/block_based/block_based_table_reader_test.cc:444]: Oh - good catch! Will fix

Comment (hx235) [reply table/block_based/block_based_table_reader_test.cc]: Fixed.

**Pull request closed without merging**